hub/src/main/java/ch/prevo/open/hub/match/Matcher.java

```diff
@@ -0,0 +1,40 @@
+package ch.prevo.open.hub.match;
+
+import ch.prevo.open.encrypted.model.InsurantInformation;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class Matcher {
+
+    public List<Match> findMatches(Set<InsurantInformation> retirementFundExits, Set<InsurantInformation> retirementFundEntries) {
+        List<Match> matches = new ArrayList<>();
+        for (InsurantInformation exit : retirementFundExits) {
+            Match match = findMatchingEntry(retirementFundEntries, exit);
+            if (match != null) {
+                matches.add(match);
+            }
+
+        }
+        return matches;
+    }
+
+    private Match findMatchingEntry(Set<InsurantInformation> retirementFundEntries, InsurantInformation exit) {
```

Comment:
Wouldn't that be perfect to use `Optional`?

Reply:
Changed the code differently: `findMatchingEntry` now returns the found entry (if any); here, an optional is less readable.